### PR TITLE
[PR-20964] Add Caching for access token for 20 minutes

### PR DESCRIPTION
**Pull request by bohdan-perekhrest**

### CODE REVIEW

File: lib/exact_target_sdk/client.rb [hunk 1]
```diff
@@ -328,6 +328,7 @@ def soap_instance_url
   end
 
   def request_access_token
+    return if @access_token = Rails.cache.fetch(access_token_cache_key(config[:username]), nil)
```
Comment (zhuravel) [inline lib/exact_target_sdk/client.rb:331]: Gem should be framework agnostic. I expect the caching to be done in the product part.

Comment (bohdan-perekhrest) [reply lib/exact_target_sdk/client.rb:331]: okay

Comment (bohdan-perekhrest) [reply lib/exact_target_sdk/client.rb:331]: but, I don't understand what to cache in product part? if we need to get from this library client object, and in this library we cannot just pass cached access_token and skip requests.

Comment (zhuravel) [reply lib/exact_target_sdk/client.rb:331]: So we need to be able to pass cached access_token in the library.

**Pull request closed without merging**